src/encodings/encoding.rs

```diff
@@ -351,6 +354,68 @@ impl<T: DataType> Encoder<T> for DictEncoder<T> {
   }
 }
 
+// ----------------------------------------------------------------------
+// RLE encoding
+
+const DEFAULT_RLE_BUFFER_LEN: usize = 1024;
+
+/// RLE encoding for values.
+/// Currently is used only for data pages v2 and supports boolean types.
+pub struct RleValueEncoder<T: DataType> {
+  // Buffer with raw values that we collect,
+  // when flushing buffer they are encoded using RLE encoder
+  raw_values_buffer: Vec<T::T>
+}
+
+impl<T: DataType> RleValueEncoder<T> {
+  pub fn new() -> Self {
+    Self { raw_values_buffer: Vec::with_capacity(DEFAULT_RLE_BUFFER_LEN) }
+  }
+}
+
+impl<T: DataType> Encoder<T> for RleValueEncoder<T> {
+  #[inline]
+  fn put(&mut self, values: &[T::T]) -> Result<()> {
+    self.raw_values_buffer.extend_from_slice(values);
+    Ok(())
+  }
+
+  fn encoding(&self) -> Encoding {
+    Encoding::RLE
+  }
+
+  #[inline]
+  default fn flush_buffer(&mut self) -> Result<ByteBufferPtr> {
+    panic!("RleValueEncoder only supports BoolType");
+  }
+}
+
+impl Encoder<BoolType> for RleValueEncoder<BoolType> {
+  #[inline]
+  fn flush_buffer(&mut self) -> Result<ByteBufferPtr> {
+    let bit_width: u8 = 1;
+    let mut rle_encoder = RleEncoder::new(bit_width, DEFAULT_RLE_BUFFER_LEN);
+    // Encode data
+    for value in &self.raw_values_buffer {
+      if !rle_encoder.put(*value as u64)? {
+        break;
+      }
+    }
+
+    // Flush all encoder buffers and raw values
+    rle_encoder.flush()?;
+    self.raw_values_buffer.clear();
+
+    // Copy data into bytebuffer
+    let mut encoded_data = Vec::new();
+    let len = (rle_encoder.len() as i32).to_le();
+    let len_bytes = len.as_bytes();
+    encoded_data.extend_from_slice(len_bytes);
+    encoded_data.extend_from_slice(&rle_encoder.consume()[..]);
```

Comment:
maybe we can call `buffer()` instead of `consume()`? this also allows use to reuse the same `rle_encoder`, and avoid repeated allocation of its internal buffer.

Reply:
Yes, you are right. I will change it.